addons/hr_timesheet/models/project_project.py

```diff
@@ -44,9 +44,9 @@ def _compute_timesheet_encode_uom_id(self):
         for project in self:
             project.timesheet_encode_uom_id = project.company_id.timesheet_encode_uom_id or self.env.company.timesheet_encode_uom_id
 
-    @api.depends('analytic_account_id')
+    @api.depends('account_id')
     def _compute_allow_timesheets(self):
-        without_account = self.filtered(lambda t: not t.analytic_account_id and t._origin)
+        without_account = self.filtered(lambda t: t._origin and not t.account_id)
```

Comment:
Why are we looking at the origin?

Reply:
I don't really know why we decide to do that but I think it is to make sure we don't do the compute when the user is creating a new project since we will generate the analytic account in the project creation but if after the creation, the user decides to remove the analytic account then, in that case, we will also remove the timesheets feature. 🙂

I will check that why Alexandra (AJU) when she will come back.

Reply:
But it won't be recomputed unless the dependency triggers, which shouldn't be the case AFAIK

Reply:
The user can set the `account_id` field that's why we just want to do that compute once we know the project was already created since in the create method we will generate an account.
Otherwise, if during the project creation, the user enables the timesheets feature (normally it is the case by default) and sets himself an analytic account but finally he decides to remove the analytic account and then saves. By doing that, the timesheets feature will be disabled instead of automatically generating a new analytic account for that new project.

(I agree the use case is really uncommon :D )